libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5410,7 +5446,17 @@ void GCS_MAVLINK::initialise_message_intervals_for_stream(GCS_MAVLINK::streams i
         // found it!
         const uint16_t interval_ms = get_interval_for_stream(id);
         for (uint8_t j=0; j<entries.num_ap_message_ids; j++) {
+#if HAL_HIGH_LATENCY2_ENABLED
+            // All message intervals to 0 on a disabled high latency link
+            if (is_high_latency_link && !high_latency_link_enabled) {
+                set_ap_message_interval(entries.ap_message_ids[j], 0);
+            }
+            else {
```

Comment:
```suggestion
            } else {
```

... actually, this will be neater with a `continue` instead of an else block; you can use the normal code's `set_ap_message_interval` that way.

Reply:
Removed